contracts/AelinUpFrontDeal.sol

```diff
@@ -221,14 +222,6 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
 
         if (!dealConfig.allowDeallocation) {
             require(totalPoolShares <= _underlyingDealTokenTotal, "purchased amount over total");
-        } else if (totalPoolShares > _underlyingDealTokenTotal) {
```

Comment:
the idea here was to get rid of the precision issue by forcing the last user to have a math formula that always works but it doesnt really work. it only lets the current tests pass

Reply:
this is because there are a lot of users and whether or not the last user has a forumla that works doesn't mean someone else's deposit won't cause a previous user or the holder to have a precision issue in the future after they deposit

Reply:
yeah I just realised it can't work